delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/FileEvent.scala

```diff
@@ -337,19 +342,21 @@ object FileEvent {
 
       @nowarn("cat=unused")
       override val sseEncoder: Encoder.AsObject[FileEvent] = {
-        val context                                         = ContextValue(Vocabulary.contexts.metadata, contexts.files)
-        val metadataKeys: Set[String]                       =
+        val context                                                      = ContextValue(Vocabulary.contexts.metadata, contexts.files)
+        val metadataKeys: Set[String]                                    =
           Set("subject", "types", "source", "project", "rev", "instant", "digest", "mediaType", "attributes", "bytes")
-        implicit val circeConfig: Configuration             = Configuration.default
+        implicit val circeConfig: Configuration                          = Configuration.default
           .withDiscriminator(keywords.tpe)
           .copy(transformMemberNames = {
             case "id"                                  => "_fileId"
             case "subject"                             => nxv.eventSubject.prefix
             case field if metadataKeys.contains(field) => s"_$field"
             case other                                 => other
           })
-        implicit val subjectEncoder: Encoder[Subject]       = IriEncoder.jsonEncoder[Subject]
-        implicit val projectRefEncoder: Encoder[ProjectRef] = IriEncoder.jsonEncoder[ProjectRef]
+        implicit val subjectEncoder: Encoder[Subject]                    = IriEncoder.jsonEncoder[Subject]
+        implicit val projectRefEncoder: Encoder[ProjectRef]              = IriEncoder.jsonEncoder[ProjectRef]
+        implicit val userMetadataCodec: Codec.AsObject[FileUserMetadata] =
+          deriveConfiguredCodec[FileUserMetadata]
```

Comment:
I wonder if this should be treated as a leaf node and not have configured codecs? Since it really is just a `Map[String, String]` wrapper rather than a KG thing (which I assume the custom circe config is intended for)

Reply:
That sentence wasn't clear at all - just meant we don't want to accidentally map certain custom keys / values based on config like `transformMemberNames`

Reply:
Yeah, if we're testing mapped fields as the metadata fields. Just a regular Codec would work I think, without the configured part e.g. `deriveCodec`